core/io/logger.cpp

```diff
@@ -152,7 +152,12 @@ void RotatedFileLogger::rotate_file() {
 			OS::Time time = OS::get_singleton()->get_time();
 			sprintf(timestamp, "-%04d-%02d-%02d-%02d-%02d-%02d", date.year, date.month, date.day, time.hour, time.min, time.sec);
 
-			String backup_name = base_path.get_basename() + timestamp + "." + base_path.get_extension();
+			String backup_name = base_path.get_extension();
```

Comment:
It's a bit weird to reuse `backup_name` for this test, when it's actually not meant to be saved in this String at all.
This would be much cleaner IMO:
```
String backup_name = base_path.get_basename() + timestamp;
if (bash_path.get_extension() != String()) {
	backup_name += "." + base_path.get_extension();
}
```